javaslang/src/main/java/javaslang/collection/IndexedSeq.java

```diff
@@ -220,19 +221,30 @@ default int lastIndexWhere(Predicate<? super T> predicate, int end) {
 
     @Override
     default Iterator<T> reverseIterator() {
-        return new AbstractIterator<T>() {
-            private int i = IndexedSeq.this.length();
+        return new AbstractIterator<T>() { /*@formatter:off*/
+            final ListIterator<T> delegate = IndexedSeq.this.listIterator(IndexedSeq.this.length());
+            @Override public boolean hasNext() { return delegate.hasPrevious(); }
+            @Override public T getNext()       { return delegate.previous(); }
```

Comment:
I'm not quite sure what I'm reading here, but shouldn't there be a few more methods overridden here? 

ie nextIndex, hasPrevious, getPrevious, previousIndex